### PR TITLE
Add job statistics for CRI job environment

### PR DESCRIPTION
**Pull request by gritukan**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

### CODE REVIEW

File: yt/yt/library/containers/cgroup.cpp [hunk 1]
```diff
@@ -730,13 +730,21 @@ std::map<TString, TString> ParseProcessCGroups(const TString& str)
...
 {
     auto cgroupsPath = Format("/proc/%v/cgroup", pid);
     auto rawCgroups = TFileInput{cgroupsPath}.ReadAll();
     return ParseProcessCGroups(rawCgroups);
 }
 
+THashMap<TString, TString> GetSelfProcessCGroups()
+{
+    auto rawCgroups = TFileInput{"/proc/self/cgroup"}.ReadAll();
+    const static NLogging::TLogger Logger("XXX");
```
Comment (Kontakter) [inline yt/yt/library/containers/cgroup.cpp:743]: Some debug things, let remove it

File: yt/yt/server/job_proxy/job_proxy.cpp [hunk 2]
```diff
@@ -1005,6 +1005,36 @@ TStatistics TJobProxy::GetEnrichedStatistics() const
         } catch (const std::exception& ex) {
             YT_LOG_ERROR(ex, "Unable to get block IO statistics from resource controller");
         }
+
+        try {
+            auto jobCpuStatistics = environment->GetJobCpuStatistics()
+                .ValueOrThrow();
+            if (jobCpuStatistics) {
+                statistics.AddSample("/job/cpu", *jobCpuStatistics);
```
[the remark this answers is not in the document]
Comment (Kontakter) [reply yt/yt/server/job_proxy/job_proxy.cpp:1013]: Hm, I get it after reading TODO message above

File: yt/yt/server/job_proxy/environment.cpp [hunk 3]
```diff
@@ -954,6 +1016,39 @@ class TCriJobProxyEnvironment
...
     const TCriJobEnvironmentConfigPtr Config_;
+
+    NCGroups::TSelfCGroupsStatisticsFetcher StatisticsFetcher_;
+
+    std::optional<TJobEnvironmentMemoryStatistics> DoGetJobMemoryStatistics() const
+    {
+        auto statistics = StatisticsFetcher_.GetMemoryStatistics();
+        // NB: PeakRss usage is intentional here, to make statistics more sane.
+        return TJobEnvironmentMemoryStatistics{
+            .Rss = statistics.PeakRss,
```
Comment (Kontakter) [inline yt/yt/server/job_proxy/environment.cpp:1027]: Maybe call it as PeakRss in this case?

File: yt/yt/library/containers/cgroups_new.cpp [hunk 4]
```diff
@@ -0,0 +1,271 @@
...
+        }
+        if (tokens[1] == "memory") {
+            auto cgroup = tokens[2];
+            if (NFS::Exists(Format("/sys/fs/cgroup/memory/%v/memory.stat", cgroup))) {
+                CGroup_ = cgroup;
+                IsV2_ = false;
+                return;
+            } else if (NFS::Exists("/sys/fs/cgroup/memory/memory.stat")) {
+                CGroup_ = "/";
+                IsV2_ = false;
```
Comment (Kontakter) [inline yt/yt/library/containers/cgroups_new.cpp:251]: Here false in both cases?

File: yt/yt/server/job_proxy/job_proxy.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (gritukan) [reply yt/yt/server/job_proxy/job_proxy.cpp:1013]: It's is a cumulative resource usage of user job and job proxy. Feel free to suggest some better naming here :)

File: yt/yt/server/job_proxy/environment.cpp (hunk 3 above)
Comment (gritukan) [reply yt/yt/server/job_proxy/environment.cpp:1027]: I'd prefer to do it in a separate PR with some additional statistics refactorings.

File: yt/yt/library/containers/cgroups_new.cpp [hunk 5]
```diff
@@ -0,0 +1,267 @@
...
+#include <yt/yt/core/misc/fs.h>
+
+#include <util/stream/file.h>
+#include <util/string/vector.h>
+
+namespace NYT::NContainers::NCGroups {
+
+////////////////////////////////////////////////////////////////////////////////
+
+static inline const NLogging::TLogger Logger("CGroups");
```
Comment (k-pogorelov) [inline yt/yt/library/containers/cgroups_new.cpp:16]: inline has no affect here

**Pull request closed without merging**